packages/ds-ext/src/extension.ts

```diff
@@ -55,8 +55,8 @@ export function activate(context: vscode.ExtensionContext) {
 
   vscode.commands.registerCommand('ds-ext.showDiagramPreview', async (args: vscode.Uri) => {
     const diagramDocument = diagramEditorProvider.provideDiagramContent(args);
-    const diagramData = diagramDocument?.data;
-    const dataString = JSON.stringify(JSON.parse(new TextDecoder().decode(diagramData)), null, 2);
+    const diagramData = new TextDecoder().decode(diagramDocument?.data) || '{}';
```

Comment:
Should we consider doing something like JSON.stringify(new Diagram()) as {} is not a valid Diagram?

Reply:
I recommend using '' to replace {} or `new Diagram`, since the preview cannot edit the user's file directly.
```suggestion
     const diagramData = new TextDecoder().decode(diagramDocument?.data);
    let dataString = '';
    if (diagramData) {
      dataString = JSON.stringify(JSON.parse(diagramData), null, 2);
    }
```
Do you think that when we create a .ds file, we should set the default content to be of Diagram type? And if it's an empty .ds file, should we set it to `new Diagram`?

Reply:
Right, this is a bit more complex than I thought🙂 empty string or keep it as {} is good enough for now.